test/TestPath.hs

```diff
@@ -25,24 +26,23 @@ tests = [
 
 testPathRoundtrip =
   do
-    rt "empty path" []
-    rt "top-level path" ["foo"]
-    rt "nested path" ["foo_1", "bar_2"]
+    rt "empty path" root
+    rt "top-level path" $ Path ["foo"]
+    rt "nested path" $ Path ["foo_1", "bar_2"]
   where
     rt s path = assertEqual s (fromString $ toString path) (Just path)
 
 testPathFromString =
   do
     assertFailed "empty" (fromString "")
-    fromString "/" >>= assertEqual "slash" []
-    fromString "/foo" >>= assertEqual "single" ["foo"]
-    fromString "/foo/bar" >>= assertEqual "multiple" ["foo", "bar"]
+    fromString "/" >>= assertEqual "slash" root
+    fromString "/foo" >>= assertEqual "single" (Path ["foo"])
+    fromString "/foo/bar" >>= assertEqual "multiple" (Path ["foo", "bar"])
     -- FIXME: should the following strictly be an error?
     -- assertFailed "trailing" (fromString "/foo/bar/")
     assertFailed "no leading" (fromString "foo/bar")
 
-
 testUp =
   do
     assertEqual "root up failed" root $ up root
-    assertEqual "normal up failed" ["a"] $ up ["a", "b"]
+    assertEqual "normal up failed" (Path ["a"]) $ up $ Path ["a", "b"]
```

Comment:
I know it's not a huge deal, because you've basically it will be covered everywhere, but a quick test of the quasiquoter itself should be easy to write?

Reply:
Done.

Reply:
:tada: